OData/src/System.Web.OData/OData/ODataSwaggerConverter.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Web.Http;
+using Microsoft.OData.Edm;
+using Newtonsoft.Json.Linq;
+
+namespace System.Web.OData
+{
+    /// <summary>
+    /// Represents an <see cref="ODataSwaggerConverter"/> used to converter an Edm model to Swagger model.
+    /// </summary>
+    public class ODataSwaggerConverter
```

Comment:
E2E test case to show how customers can create a controller or action to return swagger spec, when a request uri is:
~/service.root/swagger.json, or ~/service.root/$swagger.